tiled/adapters/arrow.py

```diff
@@ -0,0 +1,322 @@
+                    )
+                ],
+            )
+        ]
+
+    #
+    @classmethod
+    def from_single_file(
+        cls,
+        data_uri: str,
+        structure: Optional[TableStructure] = None,
+        metadata: Optional[JSON] = None,
+        specs: Optional[List[Spec]] = None,
+        access_policy: Optional[AccessPolicy] = None,
+    ) -> "ArrowAdapter":
+        """
+
+        Parameters
+        ----------
+        data_uri :
+        structure :
+        metadata :
+        specs :
+        access_policy :
+
+        Returns
+        -------
+
+        """
+        return cls(
+            [data_uri],
+            structure=structure,
+            metadata=metadata,
+            specs=specs,
+            access_policy=access_policy,
+        )
+
+    def __getitem__(self, key: str) -> ArrayAdapter:
+        """
+
+        Parameters
+        ----------
+        key :
+
+        Returns
+        -------
+
+        """
+        # Must compute to determine shape.
+        return ArrayAdapter.from_array(self.read([key])[key].values)
+
+    def items(self) -> Iterator[Tuple[str, ArrayAdapter]]:
+        """
+
+        Returns
+        -------
+
+        """
+        yield from (
+            (key, ArrayAdapter.from_array(self.read([key])[key].values))
+            for key in self._structure.columns
+        )
+
+    @property
+    def reader_handle(self) -> List[pyarrow.RecordBatchFileReader]:
+        """
+        Function to initialize and return the reader handle.
+        Returns
+        -------
+        The reader handle.
+        """
+        partitions = []
+        for path in self._partition_paths:
+            if not Path(path).exists():
+                partition = None
+            else:
+                with pyarrow.ipc.open_file(path) as reader:
+                    partition = reader
+            partitions.append(partition)
+        return partitions
+
+    def write_partition(
+        self,
+        data: Union[List[pyarrow.record_batch], pyarrow.record_batch, pandas.DataFrame],
+        partition: int,
+    ) -> None:
+        """
+        "Function to write the data into specific partition as arrow format."
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        partition: integer index of partition to be read.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        uri = self._partition_paths[partition]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in batches:
+                file_writer.write_batch(batch)
+            file_writer.close()
```

Comment:
This line can be removed because it serves no purpose. The context manager guarantees that the file will be closed:

```py
with pyarrow.ipc.new_file(uri, schema) as file_writer:
    ...
# When the interpreter exits the `with ...` code block, file_write.close() is called.
```